libs/core/langchain_core/messages/ai.py

```diff
@@ -47,6 +48,87 @@ def __add__(self, other: Any) -> BaseMessageChunk:  # type: ignore
                     "Cannot concatenate AIMessageChunks with different example values."
                 )
 
+            content = merge_content(self.content, other.content)
+            additional_kwargs = merge_dicts(
+                self.additional_kwargs, other.additional_kwargs
+            )
+            response_metadata = merge_dicts(
+                self.response_metadata, other.response_metadata
+            )
+
+            if isinstance(other, AIToolCallsMessageChunk):
+                return AIToolCallsMessageChunk(
+                    example=self.example,
+                    content=content,
+                    additional_kwargs=additional_kwargs,
+                    response_metadata=response_metadata,
+                    tool_call_chunks=other.tool_call_chunks,
+                    id=self.id,
+                )
+
+            return self.__class__(
+                example=self.example,
+                content=content,
+                additional_kwargs=additional_kwargs,
+                response_metadata=response_metadata,
+                id=self.id,
+            )
+
+        return super().__add__(other)
+
+
+class ToolCall(Serializable):
+    name: str
+    args: dict
+    id: Optional[str] = None
+    index: Optional[int] = None
+
+
+class ToolCallChunk(Serializable):
+    name: Optional[str] = None
+    args: Optional[str] = None
+    id: Optional[str] = None
+    index: Optional[int] = None
+
+
+class AIToolCallsMessage(AIMessage):
+    tool_calls: Optional[List[ToolCall]] = None
+    type: Literal["tool_calls"] = "tool_calls"  # type: ignore[assignment] # noqa: E501
+
+
+class AIToolCallsMessageChunk(AIToolCallsMessage, AIMessageChunk):
+    # Ignoring mypy re-assignment here since we're overriding the value
+    # to make sure that the chunk variant can be discriminated from the
+    # non-chunk variant.
+    type: Literal["AIToolCallsMessageChunk"] = "AIToolCallsMessageChunk"  # type: ignore[assignment] # noqa: E501
+    tool_call_chunks: Optional[List[ToolCallChunk]] = None
```

Comment:
i wonder if this should be Optional[Union[List[ToolCall], List[ToolCallChunk]]]

i imagine there may be some providers down the line that only stream a completed tool call (with args already as a dict), but they will yield them one at a time

Reply:
`AIToolCallsMessageChunk` inherits `tool_calls`, can we put them there? That is what I did for Anthropic.